pkg/bpf/collection.go

```diff
@@ -41,6 +45,110 @@ func LoadCollectionSpec(path string) (*ebpf.CollectionSpec, error) {
 	return spec, nil
 }
 
+func removeUnreachableTailcalls(spec *ebpf.CollectionSpec) error {
+	type TailCall struct {
+		referenced bool
+		visited    bool
+		spec       *ebpf.ProgramSpec
+	}
+
+	entrypoints := make([]*ebpf.ProgramSpec, 0)
+	tailcalls := make(map[uint32]*TailCall)
+
+	for _, prog := range spec.Programs {
+		if prog.SectionName == "tc" || prog.SectionName == "xdp" {
+			entrypoints = append(entrypoints, prog)
+			continue
+		}
+
+		var id, slot uint32
+		if _, err := fmt.Sscanf(prog.SectionName, "%d/%v", &id, &slot); err != nil {
```

Comment:
I think this can scan less than you think. It probably makes sense to check the first return value.

> It returns the number of items successfully parsed.

Reply:
I don't understand. You want to check that the first return value == 2? I assumed this would work given I copied it from `iproute2Compat`.

Reply:
Never mind, I was thinking of C `sscanf` semantics: https://go.dev/play/p/eAbk0sJrNeI